File: backend/src/main/java/com/printit/backend/dto/staff/StaffDashboardResponse.java
package com.printit.backend.dto.staff;

import java.util.List;

public class StaffDashboardResponse {

    private long pendingOrders;
    private long printingOrders;
    private long readyForPickupOrders;
    private long completedToday;
    private List<StaffOrderSummaryResponse> recentOrders;

    public StaffDashboardResponse(
            long pendingOrders,
            long printingOrders,
            long readyForPickupOrders,
            long completedToday,
            List<StaffOrderSummaryResponse> recentOrders
    ) {
        this.pendingOrders = pendingOrders;
        this.printingOrders = printingOrders;
        this.readyForPickupOrders = readyForPickupOrders;
        this.completedToday = completedToday;
        this.recentOrders = recentOrders;
    }

    public long getPendingOrders() {
        return pendingOrders;
    }

    public long getPrintingOrders() {
        return printingOrders;
    }

    public long getReadyForPickupOrders() {
        return readyForPickupOrders;
    }

    public long getCompletedToday() {
        return completedToday;
    }

    public List<StaffOrderSummaryResponse> getRecentOrders() {
        return recentOrders;
    }
}

File: mobile/app/src/main/java/com/printit/mobile/features/staff/dashboard/StaffDashboardResponse.kt
package com.printit.mobile.features.staff.dashboard

import com.google.gson.annotations.SerializedName

data class StaffDashboardResponse(
    @SerializedName("pendingOrders")
    val pendingOrders: Long?,

    @SerializedName("printingOrders")
    val printingOrders: Long?,

    @SerializedName("readyForPickupOrders")
    val readyForPickupOrders: Long?,

    @SerializedName("completedToday")
    val completedToday: Long?,

    @SerializedName("recentOrders")
    val recentOrders: List<StaffRecentOrderResponse>?
)

data class StaffRecentOrderResponse(
    @SerializedName("id")
    val id: Long?,

    @SerializedName("fileName")
    val fileName: String?,

    @SerializedName("studentName")
    val studentName: String?,

    @SerializedName("copies")
    val copies: Long?,

    @SerializedName("status")
    val status: String?
)